### PR TITLE
[systemtest] remove systemtest in favor of bazel + opentitantool

### LINKED ISSUE #12446: [systemtest] remove (and replace with bazel)
After completing #12445, the system test python test framework can be removed and permanently replaced by bazel:

- [x] remove system test CI jobs
- [x] remove system test completely from the repository

### PR DESCRIPTION
**Pull request by timothytrippel**
Now that bazel/opentitantool are running all CI chip-level tests, systemtest may be removed. This will enable us to remove meson (which systemtest relies on).

This fixes #12446.

Note: this depends on #12925.

### CODE REVIEW

File: sw/device/lib/testing/test_framework/chip_level_test_infra.svg [hunk 1]
```diff
@@ -1,12 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
 <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
-<svg xmlns:xl="http://www.w3.org/1999/xlink" xmlns:dc="http://purl.org/dc/elements/1.1/" version="1.1" xmlns="http://www.w3.org/2000/svg" viewBox="481 186 741 502" width="741" height="502">
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:xl="http://www.w3.org/1999/xlink" viewBox="496.5145 203.5 709.1695 469.228" width="709.1695" height="469.228">
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply sw/device/lib/testing/test_framework/chip_level_test_infra.svg:3]: done

File: azure-pipelines.yml [hunk 2]
```diff
@@ -263,34 +224,12 @@ jobs:
...
-
-# TODO: change this name and replace sw_build_englishbreakfast and execute_verilated_tests_englishbreakfast
-# once testing shows it's stable
+# Software targeting the English Breakfast top level is produced by patching
+# the source tree before building. This builds a selected subset of software
+# only.
+# TODO: This is a rather ugly hack, which will go away once we properly support
+# building more than one top-level design with different parametrizations.
+# Work towards this goal is tracked in issue #4669.
 - job: bazel_build_and_execute_verilated_tests_englishbreakfast
```
Comment (milesdai) [inline azure-pipelines.yml:233]: Change this job name to `build_and_execute_verilated_tests_englishbreakfast` and make the corresponding change in the dependency names under the `chip_englishbreakfast_cw305` job

**Review by milesdai — APPROVED** [review]
LGTM

I added a comment about renaming the job, but we can also defer that to later when we subsequently do a more thorough clean up of the `azure-pipelines.yml` file if it avoids re-running this through CI again.

File: azure-pipelines.yml (hunk 2 above)
Comment (timothytrippel) [reply azure-pipelines.yml]: ah right. thanks. fixed.

**Comment by timothytrippel** [comment]
PR dependency removed. Ready for merge after CI passes.

**Pull request merged**